truffle/src/main/java/org/jruby/truffle/core/exception/SystemCallErrorLayout.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2015, 2016 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.jruby.truffle.core.exception;
+
+import com.oracle.truffle.api.object.DynamicObject;
+import com.oracle.truffle.api.object.DynamicObjectFactory;
+import com.oracle.truffle.api.object.dsl.Layout;
+import com.oracle.truffle.api.object.dsl.Nullable;
+import org.jruby.truffle.language.backtrace.Backtrace;
+
+@Layout
+public interface SystemCallErrorLayout extends ExceptionLayout {
+
+    DynamicObjectFactory createSystemCallErrorShape(
+        DynamicObject logicalClass,
+        DynamicObject metaClass);
+
+    DynamicObject createSystemCallError(
+        DynamicObjectFactory factory,
+        @Nullable Object message,
+        @Nullable Backtrace backtrace,
+        @Nullable Object errno);
```

Comment:
@nirvdrum I think it will only be int or null, should `@Nullable Integer` work? I think I was having trouble trying to use `@Nullable Integer`.

https://github.com/ruby/ruby/blob/trunk/error.c#L1486